bitcoin/chainparams.c

```diff
@@ -0,0 +1,73 @@
+#include "chainparams.h"
+#include <string.h>
+
+#define CHAINPARAMS_NETWORK_COUNT 4
+
+const struct chainparams networks[CHAINPARAMS_NETWORK_COUNT] = {
+    {.index = 0,
```

Comment:
I just wanted to have all this info in a single struct, and not having to pass the index around externally. This way we just add a pointer to the struct and if we need to have a short reference to pass to subdaemons we can just pass the index.